gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/model/builder/KspModelBuilder.kt

```diff
@@ -33,11 +32,9 @@ class KspModelBuilder : ToolingModelBuilder {
         return modelName == Ksp::class.java.name
     }
 
-    override fun buildAll(modelName: String, project: Project): Any? {
-        if (modelName == Ksp::class.java.name) {
-            val extension = project.extensions.getByType(KspExtension::class.java)
-            return KspImpl(project.name)
+    override fun buildAll(modelName: String, project: Project): Any =
+        when (modelName) {
+            Ksp::class.java.name -> KspImpl(project.name)
+            else -> error("Unknown model name: $modelName")
```

Comment:
I don't think this can ever happen due to the method above. It had to change because the java method requires a non null value.